src/firebase_functions/params.py

```diff
@@ -291,52 +330,68 @@ def value(self) -> str:
 class IntParam(Param[int]):
     """A parameter as a int value."""
 
+    @property
     def value(self) -> int:
         if _os.environ.get(self.name) is not None:
             return int(_os.environ[self.name])
         if self.default is not None:
-            return self.default
+            return self.default.value if isinstance(
+                self.default, Expression) else self.default
         return int()
 
 
 @_dataclasses.dataclass(frozen=True)
 class FloatParam(Param[float]):
     """A parameter as a float value."""
 
+    @property
     def value(self) -> float:
         if _os.environ.get(self.name) is not None:
             return float(_os.environ[self.name])
         if self.default is not None:
-            return self.default
+            return self.default.value if isinstance(
+                self.default, Expression) else self.default
         return float()
 
 
 @_dataclasses.dataclass(frozen=True)
 class BoolParam(Param[bool]):
     """A parameter as a bool value."""
 
+    @property
     def value(self) -> bool:
         env_value = _os.environ.get(self.name)
         if env_value is not None:
-            if env_value.lower() in ["true", "t", "1", "y", "yes"]:
-                return True
-            if env_value.lower() in ["false", "f", "0", "n", "no"]:
-                return False
-            raise ValueError(f"Invalid value for {self.name}: {env_value}")
+            return env_value.lower() == "true"
         if self.default is not None:
-            return self.default
+            return self.default.value if isinstance(
+                self.default, Expression) else self.default
         return False
 
 
 @_dataclasses.dataclass(frozen=True)
 class ListParam(Param[list]):
```

Comment:
Sorry - after all this work, I think the team is considering yanking ListParam support all together 🤦🏼 .

It isn't documented in https://firebase.google.com/docs/functions/config-env.

It is documented in JS SDK reference doc, and the team feels that it was a mistake to release it prematurely :'(